reference-architecture/azure-ansible/bastion.json

```diff
@@ -0,0 +1,251 @@
+          "vmSize" : "[parameters('vmSize')]"
+        },
+        "osProfile" : {
+          "computerName" : "[parameters('vmName')]",
+          "adminUsername" : "[parameters('adminUsername')]",
+          "linuxConfiguration" : {
+            "disablePasswordAuthentication" : "true",
+            "ssh" : {
+              "publicKeys" : [
+                {
+                  "path" : "[variables('sshKeyPath')]",
+                  "keyData" : "[parameters('sshKeyData')]"
+                }
+              ]
+            }
+          }
+        },
+        "storageProfile" : {
+          "imageReference" : "[parameters('imageReference')]",
+          "osDisk" : {
+            "name" : "osdisk",
+            "vhd" : {
+              "uri" : "[concat('http://', parameters('sa'), '.blob.core.windows.net/openshiftvmachines/', parameters('vmName'), '.vhd')]"
+            },
+            "caching" : "ReadWrite",
+            "createOption" : "FromImage"
+          },
+          "dataDisks" : [
+            {
+              "caching" : "None",
+              "createOption" : "Empty",
+              "diskSizeGB" : "128",
+              "lun" : 0,
+              "name" : "datadisk1",
+              "vhd" : {
+                "uri" : "[concat('http://', parameters('sa'), '.blob.core.windows.net/openshiftvmachines/', parameters('vmName'), '_datadisk1.vhd')]"
+              }
+            }
+          ]
+        },
+        "networkProfile" : {
+          "networkInterfaces" : [
+            {
+              "id" : "[resourceId('Microsoft.Network/networkInterfaces', concat(parameters('vmName'), 'nic'))]"
+            }
+          ]
+        }
+      }
+    },
+    {
+      "type" : "Microsoft.Compute/virtualMachines/extensions",
+      "name" : "[concat(parameters('vmName'),'/installcustomscript')]",
+      "apiVersion" : "[variables('apiVersion')]",
+      "location" : "[resourceGroup().location]",
+      "tags" : {
+        "displayName" : "VirtualMachineCustomScriptExtension"
+      },
+      "dependsOn" : [
+        "[concat('Microsoft.Compute/virtualMachines/', parameters('vmName'))]"
+      ],
+      "properties" : {
+        "publisher" : "Microsoft.OSTCExtensions",
+        "type" : "CustomScriptForLinux",
+        "typeHandlerVersion" : "1.3",
+        "settings" : {
+          "fileUris" : [
+            "[concat(parameters('baseTemplateUrl'), 'bastion.sh')]"
+          ],
+          "commandToExecute" : "[ concat('sh bastion.sh ', resourceGroup().name, ' ', parameters('adminUsername'), ' ', parameters('adminPassword'), ' ', reference(concat(parameters('vmName'), 'pip')).dnsSettings.fqdn, ' ', padLeft(parameters('numberOfNodes'), 2, '0'), ' ', parameters('routerExtIP'),' ', parameters('RHNUserName'), ' ', parameters('RHNPassword'),' ', parameters('SubscriptionPoolId'), ' ',  variables('sQuote'),  parameters('sshPrivateData'), variables('sQuote'),  ' ', variables('sQuote'), parameters('sshKeyData'), variables('sQuote')  ) ]"
```

Comment:
Probably just a small NIT but maybe change the var to RHSM rather than RHN. It should make retro fitting to common plays easier.